File: find-bottom-left-tree-value.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int findBottomLeftValue(TreeNode* root) {
        queue<TreeNode*> row;
        TreeNode* curr;
        
        for(row.push(root); !row.empty();){
            curr = row.front();
            for(int size = row.size(), i = 0; i < size; i++, row.pop()){
                if(row.front()->left)
                    row.push(row.front()->left);
                if(row.front()->right)
                    row.push(row.front()->right);
            }
        }

        return curr->val;
    }
};

File: find-bottom-left-tree-value.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public int FindBottomLeftValue(TreeNode root) {
        Queue<TreeNode> row = new Queue<TreeNode>();
        TreeNode curr = null;
        
        for(row.Enqueue(root); row.Count() != 0;){
            curr = row.Peek();
            for(int size = row.Count(), i = 0; i < size; i++, row.Dequeue()){
                if(row.Peek().left != null)
                    row.Enqueue(row.Peek().left);
                if(row.Peek().right != null)
                    row.Enqueue(row.Peek().right);
            }
        }

        return curr.val;
    }
}